lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -577,9 +586,14 @@ def _on_opensearch_data_storage_detaching(self, _: StorageDetachingEvent):  # no
                     if node.name != self.unit_name
                 ]
                 self._compute_and_broadcast_updated_topology(remaining_nodes)
-            elif self.app.planned_units() == 0 and self.model.get_relation(PeerRelationName):
-                self.peers_data.delete(Scope.APP, "bootstrap_contributors_count")
-                self.peers_data.delete(Scope.APP, "nodes_config")
+            elif self.app.planned_units() == 0:
```

Comment:
@smethnani I am guessing this is just a "guard rail"? B/c storage-detach runs as the very last event in the target units (the ones going away) and there is a very high chance the remote units already got the -broken/-departed events by now.

I do think we should keep this logic, but I think only on really busy clusters, where a bunch of events are running at once and remote units may not have received the -departed yet would be really make use of this logic.